packages/keymaps/src/browser/keymaps-service.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { FrontendApplicationContribution, FrontendApplication } from '@theia/core/lib/browser';
+import { Disposable, DisposableCollection, CommandRegistry, KeybindingRegistry, ILogger, ResourceProvider, Resource } from '@theia/core/lib/common';
+import { RawKeybinding, KeybindingScopes } from '@theia/core/lib/common';
+import { UserStorageUri } from '@theia/userstorage/lib/browser/';
+import URI from '@theia/core/lib/common/uri';
+import * as ajv from 'ajv';
+
+export const keymapsSchema = {
+    "type": "array",
+    "properties": {
+        "keybinding": {
+            "type": "string"
+        },
+        "command": {
+            "type": "string"
+        },
+        "context": {
+            "type": "string"
+        },
+    },
+    "required": ["command", "keybinding"],
+    "optional": [
+        "context"
+    ],
+    "additionalProperties": false
+};
+
+export const keymapsUri: URI = new URI('keymaps.json').withScheme(UserStorageUri.SCHEME);
+
+@injectable()
+export class KeymapsService implements Disposable, FrontendApplicationContribution {
```

Comment:
What about the workspace scope? Is it out of the scope of this PR?

Reply:
Yea I was thinking about it after (in another PR). Maybe we could extract a generic class from keymaps-service that could be implemented by both `KeymapsServiceUser` and `KeymapsServiceWorkspace`.